taf/testlib/virtual_env.py

```diff
@@ -420,6 +442,30 @@ def _reuse_public_access(self, external_net, routers_client=None, tenant_id=None
 
         return False
 
+    def _reuse_public_network(self, external_net, routers_client=None, networks_client=None, tenant_id=None):
+        """
+        @brief   Search for the external network
+        """
+        if not tenant_id:
+            return None
+
+        if not routers_client:
+            routers_client = self.handle.os_adm.routers_client
+        if not networks_client:
+            networks_client = self.handle.os_adm.networks_client
+
+        net_2_router_map = self._get_external_elements(routers_client=routers_client, tenant_id=tenant_id)
+        for network_id, router_id in net_2_router_map.items():
+            net_network = networks_client.show_network(network_id)['network']
+            for sub_net in net_network['subnets']:
+                if external_net and self.handle.os_adm.subnets_client.show_subnet(sub_net)['subnet']['gateway_ip'] in external_net:
```

Comment:
I've moved check for external_net to the beginning. I've removed router_id which were not used.
Rest remain the same as proposed solution is not working as it is. 
show_subnet needs subnet_id,  net_id does not contain 'subnet' as it is just Id. I replaced net_id with corresponding network.
Intersection provides empty set. There seems to be wrong type of object to intersect with.
I stop next further investigation here. Original solution was tested and it is working.